### PR TITLE
[4.x] Initial documentation for the new Helidon connector.

### PR DESCRIPTION
**Pull request by spericas**
### Description
Documentation for new Helidon connector in 4.x

### Documentation
Not needed, it would be meta-documentation in this case

### CODE REVIEW

File: docs/mp/jaxrs/helidon-connector.adoc [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+|(none)
+
+|jersey.connector.helidon.shareConnectionCache
+|`Boolean`
+|client
+|false
+|===
+
+
+=== HTTP/2 support
```
Comment (ljamen) [inline docs/mp/jaxrs/helidon-connector.adoc:149]: Support

File: docs/mp/jaxrs/helidon-connector.adoc [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+and the first response is HTTP/2 (binary), assuming the negotiation is successful.
+2. Similar to (1) except that a TLS extension called ALPN is used to convey the
+upgrade negotiation. Naturally, this only works with secure connections, so TLS
+is a requirement here.
+3. Using prior knowledge, where the client simply sends an HTTP/2 request knowing
+_a priori_ that the server is capable of handling it. This option always requires TLS.
+
+== Examples
+
+=== HTTP/2 negotiation without TLS
```
Comment (ljamen) [inline docs/mp/jaxrs/helidon-connector.adoc:165]: We're using initcap for H1, H2 and H3
HTTP/2 Negotiation Without TLS

File: docs/mp/jaxrs/helidon-connector.adoc [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+  }
+----
+
+NOTE: Properties in the Jakarta REST Client API can be set on `Client`, `WebTarget` and
+`Invocation`  and are inherited accordingly.
+
+The request invocation in the example above will include an HTTP/2 protocol upgrade
+request which may be granted by the server if HTTP/2 support is enabled.
+
+=== HTTP/2 negotiation with TLS/ALPN
```
Comment (ljamen) [inline docs/mp/jaxrs/helidon-connector.adoc:190]: Negotiation with TLS/ALPN

File: docs/mp/jaxrs/helidon-connector.adoc [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+  WebTarget webTarget = client.target(...).path(...);
+  try (Response response = webTarget.request().get()) {
+      // ...
+  }
+----
+
+The call to `addApplicationProtocol()` indicates the desire to negotiate a protocol upgrade. Naturally,
+ALPN only works on secure connections, so TLS is always configured at the same time.
+
+=== HTTP/2 prior knowledge
```
Comment (ljamen) [inline docs/mp/jaxrs/helidon-connector.adoc:218]: Prior Knowledge

**Review by ljamen — APPROVED** [review]
Very small edits for initcap but LGTM and should be merged.

**Comment by spericas** [comment]
@ljamen Comments addressed.

**Review by ljamen — APPROVED** [review]
LGTM

**Pull request merged**